doc/development_process/review_process.rst

```diff
@@ -0,0 +1,205 @@
+
+Categories/Labels
+-----------------
+
+Hotfix
+++++++
+
+Any change that is a fix to an issue that blocks developers from doing their
+daily work, for example CI breakage, Test breakage, Minor documentation fixes
+that impact the user experience.
+
+Such fixes can be merged at any time after they have passed CI checks. Depending
+on the fix, severity and availability of someone to review them (other than the
+author) they can be merged with justification without review by one of the
+project owners.
+
+Trivial
++++++++
+
+Generic changes that appear obvious enough and do not require maintainer or code-owner
+involvement. Such changes should not change the logic or the design of a
+subsystem or component. For example a trivial change can be:
+
+- Documentation changes
+- Configuration changes
+- Minor Build System tweaks
+- Minor optimization to code logic without changing the logic
+- Test changes and fixes
+- Sample modifications to support additional configuration or boards etc.
+
+Maintainer
++++++++++++
+
+Any changes that touch the logic or the original design of a subsystem or
+component will need to be reviewed by the code owner or the designated subsystem
+maintainer. If the code changes is initiated by a contributor or developer other
+than the owner the pull request needs to be assigned to the code owner who will
+have to drive the pull request to a mergeable state by giving feedback to the
+author and asking for more reviews from other developers.
+
+Security
++++++++++++
+
+Changes that appear to have an impact to the overall security of the system need
+to be reviewed by a security expert from the security working group.
+
+TSC
+++++
+
+Changes that introduce new features or functionality or change the way the
+overall system works need to be reviewed by the TSC or the responsible Working
+Group. For example for API changes, the API working group needs to be consulted
+and made aware of such changes.
+
+
+A Pull-Request should have an Assignee
+=======================================
+
+- An assignee to a pull request should not be the same as the
+  author of the pull-request
+- An assignee to a pull request is responsible for driving the
+  pull request to a mergeable state
+- An assignee is responsible for dismissing stale reviews and seeking reviews
+  from additional developers and contributors
+- Pull requests should not be merged without an approval by the assignee.
+
+Pull Request should not be merged by author without review
+===========================================================
+
+All pull requests need to be reviewed and should not be merged by author
+without a review. The following exceptions apply:
+
+- Hot fixes: Fixing CI issues, reverts and system breakage
+- Release related changes: Changing version file, applying tags and release
+  related activities without any code changes.
+
+Developers and contributors should always seek review, however there are cases
```

Comment:
The audit folks would likely take exception to rushed merges without sufficient reviews...